crypto/src/lib.rs

```diff
@@ -489,35 +411,26 @@ impl From<PublicKey> for Multihash {
 
 ffi::ffi_item! {
     /// Private Key used in signatures.
-    #[derive(DebugCustom, Display, Clone, PartialEq, Eq, Deserialize, Serialize, FfiType)]
-    #[debug(fmt = "{{ digest: {digest_function}, payload: {payload:X?}}}")]
-    #[display(fmt = "{}", "hex::encode(payload)")]
-    #[allow(clippy::multiple_inherent_impl)]
+    #[derive(Clone, PartialEq, Eq, CopyGetters)]
+    #[cfg_attr(not(feature="ffi_import"), derive(derive_more::DebugCustom, Display, Deserialize, Serialize))]
+    #[cfg_attr(not(feature="ffi_import"), debug(fmt = "{{ digest: {digest_function}, payload: {payload:X?}}}"))]
+    #[cfg_attr(not(feature="ffi_import"), display(fmt = "{}", "hex::encode(payload)"))]
```

Comment:
This seems to be unique to the type, and IMO should be extended to a few more types.

Reply:
can you elaborate what do you mean? yes, lots of types will have this guard

Reply:
This **should** apply to other types, but right now only applies to this.

Reply:
it's still not clear to me what you mean here. Can you point me to one such type that would require **this**?